packages/compass-shell/src/modules/history-storage.spec.js

```diff
@@ -1,49 +1,49 @@
-/* eslint-disable no-sync */
 import path from 'path';
 import os from 'os';
 import { expect } from 'chai';
-import fs from 'fs';
+import fs from 'fs/promises';
 
 import { HistoryStorage } from './history-storage';
 
 describe('HistoryStorage', function () {
-  let tempDir;
+  let tmpDir;
   let historyFilePath;
 
-  beforeEach(function () {
-    tempDir = path.join(os.tmpdir(), `compass-shell-test-${Date.now()}`);
-    historyFilePath = path.join(tempDir, 'shell-history.json');
+  let historyStorage;
+
+  beforeEach(async function () {
+    tmpDir = await fs.mkdtemp(path.join(os.tmpdir(), 'compass-shell-test'));
+    historyFilePath = path.join(tmpDir, 'shell-history.json');
+
+    historyStorage = new HistoryStorage(tmpDir);
   });
 
-  afterEach(function (done) {
-    fs.rm(tempDir, { recursive: true, force: true }, done);
+  afterEach(async function () {
+    await fs.rm(tmpDir, { recursive: true });
   });
 
   describe('#save', function () {
     it('creates the file and directory if not existing', async function () {
-      expect(fs.existsSync(historyFilePath)).to.be.false;
+      expect(async () => await fs.access(historyFilePath)).to.throw;
```

Comment:
This is called, but not awaited, and so the expectation will always return true. It will be caught as unhandled rejection, but this relies on how we configure mocha, not the actual assertion. A good way of testing async errors like that is do use `try / catch` in combination with `error.fail`, you can find examples of this in the codebase, it usually looks something like this:

```js
try {
  await weExpectThisMethodToReject();
  expect.fail('Expected weExpectThisMethodToReject to throw')
} catch (err) {
  expect(err).to.have.property('message', 'Expected error') // ... or any other assertion that this is an error we are expecting to see
}
```

Reply:
yes, i also realized this and made the change. thanks for suggested to fail the expectation in try block.